pandas/tests/series/test_missing.py

```diff
@@ -1315,16 +1328,17 @@ def test_spline_interpolation(self):
 
     @td.skip_if_no_scipy
     def test_spline_error(self):
-        # see gh-10633
+        # see GH-10633, GH-24014
         s = pd.Series(np.arange(10) ** 2)
         s[np.random.randint(0, 9, 3)] = np.nan
-        msg = "invalid method 'spline' to interpolate"
+        msg = "You must specify the order of the spline or polynomial"
         with pytest.raises(ValueError, match=msg):
             s.interpolate(method='spline')
 
         msg = "order needs to be specified and greater than 0"
-        with pytest.raises(ValueError, match=msg):
-            s.interpolate(method='spline', order=0)
+        for invalid_order in [-1, -1., 0, 0., np.nan]:
+            with pytest.raises(ValueError, match=msg):
+                s.interpolate(method='spline', order=invalid_order)
```

Comment:
Let's separate this code into a separate test.  That way, we can also use `pytest` parameterization.

Reply:
Done.